tests/aggregation/tests.py

```diff
@@ -1500,27 +1500,30 @@ def test_aggregation_subquery_annotation_values(self):
             ],
         )
 
+    @skipUnlessDBFeature("supports_subqueries_in_group_by")
     def test_aggregation_subquery_annotation_values_collision(self):
         books_rating_qs = Book.objects.filter(
-            publisher=OuterRef("pk"),
-            price=Decimal("29.69"),
+            publisher=OuterRef("book"),
         ).values("rating")
         publisher_qs = (
             Publisher.objects.filter(
                 book__contact__age__gt=20,
-                name=self.p1.name,
             )
             .annotate(
                 rating=Subquery(books_rating_qs),
-                contacts_count=Count("book__contact"),
             )
             .values("rating")
-            .annotate(total_count=Count("rating"))
+            .annotate(total_count=Count("*"))
+            .order_by("rating")
         )
         self.assertEqual(
             list(publisher_qs),
             [
-                {"rating": 4.0, "total_count": 2},
+                {"rating": None, "total_count": 2},
```

Comment:
This is the demonstrating that publisher book row get grouped by rating annotation now like they should while previously the query would get incorrectly grouped by `publisher.id`.

I've confirmed that the adjusted test fails with an "ambiguous alias" when applied to 10866a10fe9f0ad3ffdf6f93823aaf4716e6f27c without the `or alias in column_names` branch adjustment which should demonstrate that it remains a proper regression test.